### DISCUSSION TITLE
Visualisation du lien des services applicatifs

### DISCUSSION THREAD

**Discussion by Guerisan** [dbarzin/mercator#223] [General]
Bonjour,
Déjà bravo pour cette application ! Nous étions étions précisément à la recherche d'une solution permettant de répondre aux recommandations du site de l'ANSSI.

Au niveau de la représentation des services applicatifs toutefois, il est difficile de s'y retrouver entre les applications qui "délivrent" ce service, et celles qui "l'utilisent" (Ce qui serait utile pour une étude d'impact).
Si je recherche par exemple les services encapsulés par une même application,  le tableau du service sur la vue "applications" rassemble toutes les applications impliquées, à la ligne "Liste des applications qui utilisent ce service applicatif"
C'est effectivement ce qui est représenté par le schéma en haut de page, mais je ne sais pas de quelle application fait partie ce service.
Je suis obligée de chercher chez chaque application impliquée, laquelle délivre ce service.

Est-ce qu'il existe un moyen de récupérer plus facilement l'information ?

**Comment by dbarzin** [comment]
Bonjour,
Le lien entre application et service représente la mise à disposition du service par l'application. Le lien "d'utilisation" d'un service est représenté par un flux.
Il y a une erreur dans le texte, il va être remplacé par "Liste des applications qui délivrent ce service".
Cordialement,
Didier

**Comment by Guerisan** [comment]
Merci pour votre réponse Didier.
Est-ce qu'il y a un moyen de visualiser graphiquement à la fois le lien de mise à disposition, et les flux d'utilisation ?

**Comment by dbarzin** [comment]
Pour l'instant ce n'est pas possible mais il est possible de préfixer le nom de l'application dans le nom du service afin d'identifier quelle application sert ce service.

**Reply by Guerisan** [reply]
C'est noté ! J'ai pu assez facilement rajouter des edges dans dans la vue d'exploration en éditant le controller. Je peux proposer le commit si vous le souhaitez :)

**Reply by dbarzin** [reply]
Oui, volontiers. Vous pouvez fait un PR dans la branche Dev.
Merci,
Didier

**Reply by dbarzin** [reply]
Ton PR a bien été posté mais il ne contient aucun changements.

**Reply by Guerisan** [reply]
C'est corrigé, j'avais envoyé les changements sur ma propre branche dev, mais fait un pull request depuis ma main.

**Reply by dbarzin** [reply]
C'est intégré, merci pour la contribution !

**Comment by dbarzin** [comment]
On pourrait ajouter une flèche pour identifier que c'est un flux.

**Reply by Guerisan** [reply]
Tout à fait, ils méritent une distinction visuelle.
J'essaierai de proposer quelque chose prochainement !

**Comment by dbarzin** [comment]
Si tu es chaud, la librairie vis.js utilisée pour afficher les graphes d'objets permet d'ajouter des boutons (ex: http://tiddlymap.org/).
Il serait intéressant de pouvoir :
- recentrer
- déplacer le graphe
- zoomer
- supprimer un objet de la visualisation
- ...
